it/v4/endpoints/AmendHistoricFhlUkPropertyPeriodSummaryControllerISpec.scala

```diff
@@ -0,0 +1,250 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v4.endpoints
+
+import api.models.errors._
+import api.models.utils.JsonErrorValidators
+import com.github.tomakehurst.wiremock.stubbing.StubMapping
+import play.api.http.HeaderNames.ACCEPT
+import play.api.http.Status._
+import play.api.libs.json.{JsNumber, JsObject, JsValue, Json}
+import play.api.libs.ws.{WSRequest, WSResponse}
+import play.api.test.Helpers.AUTHORIZATION
+import support.IntegrationBaseSpec
+import v3.models.request.amendHistoricFhlUkPiePeriodSummary.AmendHistoricFhlUkPiePeriodSummaryRequestBody
+import v4.stubs.{AuthStub, DownstreamStub, MtdIdLookupStub}
+
+class AmendHistoricFhlUkPropertyPeriodSummaryControllerISpec extends IntegrationBaseSpec with JsonErrorValidators {
+
+  val requestBodyJson: JsValue = Json.parse(
+    """{
+      |   "income":{
+      |      "periodAmount":1123.45,
+      |      "taxDeducted":2134.53,
+      |      "rentARoom":{
+      |         "rentsReceived":5167.56
+      |       }
+      |   },
+      |   "expenses":{
+      |      "premiseRunningCosts":5167.53,
+      |      "repairsAndMaintenance":424.65,
+      |      "financialCosts":853.56,
+      |      "professionalFees":835.78,
+      |      "costOfServices":978.34,
+      |      "other":382.34,
+      |      "travelCosts":145.56,
+      |      "rentARoom":{
+      |         "amountClaimed":945.9
+      |       }
+      |   }
+      |}
+      |""".stripMargin
+  )
+
+  private val requestBodyJsonConsolidatedExpenses = Json.parse(
+    """{
+      |   "income":{
+      |      "periodAmount":1123.45,
+      |      "taxDeducted":2134.53,
+      |      "rentARoom":{
+      |         "rentsReceived":5167.56
+      |       }
+      |   },
+      |   "expenses":{
+      |      "consolidatedExpenses":135.78
+      |    }
+      |}
+      |""".stripMargin
+  )
+
+  private trait Test {
+    val nino: String     = "AA123456A"
+    val periodId: String = "2017-04-06_2017-07-04"
+
+    def setupStubs(): StubMapping
+
+    def ifsUri: String = s"/income-tax/nino/$nino/uk-properties/furnished-holiday-lettings/periodic-summaries"
+
+    def ifsQueryParams: Map[String, String] = Map(
+      "from" -> "2017-04-06",
+      "to"   -> "2017-07-04"
+    )
+
+    def request(): WSRequest = {
+      setupStubs()
+      buildRequest(s"/uk/period/furnished-holiday-lettings/$nino/$periodId")
+        .withHttpHeaders(
+          (ACCEPT, "application/vnd.hmrc.3.0+json"),
```

Comment:
I think all the iSpecs in v4 should have a v4 accept header?
```suggestion
          (ACCEPT, "application/vnd.hmrc.4.0+json"),
```